lib/rspec/matchers/built_in/has.rb

```diff
@@ -46,8 +46,22 @@ def description
 
       private
 
+        # Catch a semi-frequent typo - if you have strict_predicate_matchers disabled and
+        # expect(spy).to have_receieveddd(:foo) it would be evergreen - the dynamic matcher
+        # queries `has_receiveddd?`, the spy _fakes_ the method, returning its (truthy) self.
+        def really_responds_to?(method)
+          if defined?(RSpec::Mocks::Double) && @actual.is_a?(RSpec::Mocks::Double)
+            @actual.respond_to?(method) && @actual.methods.include?(method)
+          else
+            @actual.respond_to?(method)
+          end
+        rescue NoMethodError
+          # Proxied BasicObjects don't have `is_a?`, or basically anything else.
+          @actual.respond_to?(method)
+        end
+
         def predicate_accessible?
-          @actual.respond_to? predicate
+          really_responds_to?(predicate)
         end
 
         # support 1.8.7, evaluate once at load time for performance
```

Comment:
The line below can suggest the solution. Were object.methods a list if stings back in the 1.8.7?

Reply:
Yup! I had no idea - I was only a young impressionable rubyist in those days, and did not use such magical methods :-D

https://rubydoc.info/stdlib/core/1.8.7/Object:methods